### PR TITLE
fixed default value of execution_timeout in base operator

### LINKED ISSUE #18578: Allow to set execution_timeout default value in airflow.cfg
### Discussed in https://github.com/apache/airflow/discussions/18411

<div type='discussions-op-text'>

<sup>Originally posted by **alexInhert** September 21, 2021</sup>
### Description

Currently the default value of execution_timeout in base operator is None
https://github.com/apache/airflow/blob/c686241f4ceb62d52e9bfa607822e4b7a3c76222/airflow/models/baseoperator.py#L502

This means that a task will run without limit till finished.

### Use case/motivation

The problem is that there is no way to overwrite this default for all dags. This causes problems where we find that tasks run sometimes for 1 week!!! for no reason. They are just stuck. This mostly happens with tasks that submit work to some 3rd party resource. The 3rd party had some error and terminated but the Airflow task for some reason did not.

This can be handled in cluster policy however it feels that cluster policy is more to enforce something but less about setting defaults values.

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)
</div>

### PR DESCRIPTION
**Pull request by Ayushsunny**
Fixed #18578

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: airflow/models/baseoperator.py [hunk 1]
```diff
@@ -502,7 +502,7 @@ def __init__(
         pool: Optional[str] = None,
         pool_slots: int = 1,
         sla: Optional[timedelta] = None,
-        execution_timeout: Optional[timedelta] = None,
+        execution_timeout: Optional[timedelta] = conf.gettimedelta('core', 'default_ execution_timeout', fallback=None),
```
Comment (alexInhert) [inline airflow/models/baseoperator.py:505]: there is no `gettimedelta` function in https://github.com/apache/airflow/blob/main/airflow/configuration.py . You need to implement it.

Comment (Ayushsunny) [reply airflow/models/baseoperator.py:505]: @alexInhert What's `gettimedelta` function supposed to do here?

Comment (alexInhert) [reply airflow/models/baseoperator.py:505]: execution_timeout accept timedelta object. You are reading value from a configuration file you need to convert it into the object.

Comment (zevaverbach) [reply airflow/models/baseoperator.py:505]: I'd like to take this task over if @Ayushsunny is okay with it?

Comment (Ayushsunny) [reply airflow/models/baseoperator.py:505]: @zevaverbach Yeah, Sir sure take it as I'm busy with my University works.
Thank You.

Comment (eladkal) [reply airflow/models/baseoperator.py:505]: @zevaverbach please comment on the issue so I can assign it to you :) 
https://github.com/apache/airflow/issues/18578

**Comment by eladkal** [comment]
Closing as @zevaverbach will take over and open a new PR.

**Pull request closed without merging**